ilastik/workflows/tracking/common.py

```diff
@@ -0,0 +1,16 @@
+import sys
+
+
+class DivisionClassifierLabelNames:
+    def __getitem__(self, idx: int) -> str:
+        assert isinstance(idx, int)
+        assert idx >= 0
+        if idx == 0:
+            return "False Detection"
+        elif idx == 1:
+            return "1 Object"
+        else:
+            return f"{idx:d} Objects"
+
+    def __len__(self) -> int:
+        return sys.maxsize
```

Comment:
This could result in `MemoryError` if someone does `list(DivisionClassifierLabelNames())`, directly or indirectly. Is it possible to remove `__len__` and modify code that does this?

Reply:
True, I also saw that problem and thought, how likely is that to happen :D. I guess the calling code could `try: ... except KeyError` instead of checking length.

Reply:
side note: so I haven't found the code for "list(...)" yet, but it seems that it only checks if __len__ gives something nonzero and then iterates over the object until it catches `KeyError`

Reply:
In this case it makes sense to choose label names at the calling site. We can add slice support to `__getitem__` to fully mimic `list`, and just use something like `LABEL_NAMES[:15]`.

Reply:
btw, even with `__len__` removed `list(..)` will dutifully run into eternity with an object like

```python

class Blah:
     def __getitem__(self, idx):
         return idx

blah = Blah()
list(blah)
```

Reply:
Interesting. It seems that we forgot that the [iterator protocol](https://docs.python.org/3/library/functions.html#iter) also works on objects with just `__getitem__`, not only on objects with `__iter__`.

What if we still add `__len__`, but `raise NotImplementedError` inside? This should prevent functions that try to create a finite sequence not to do that. OTOH, one may argue that call sites shouldn't try to create such sequences in the first place.

Reply:
Okay I had the feeling that things are over-complicated here. I pushed a small update to  that class with a max-size of 1024 (which is unreasonably large for the tracking problem).